File: Service/Model/Menu.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Service.Model
{
    [Table("Menu")]
    public class Menu
    {
        public int Id { get; set; }
        public int? ParentMenu { get; set; }
        public required string Group { get; set; }
        public required string Name { get; set; }
        public required string Alias { get; set; }
        public int? Index { get; set; }
        public bool? ShowHomePage { get; set; }
        public required string Type { get; set; }
        public bool Active { get; set; }

        [ForeignKey("ParentMenu")]
        public virtual Menu? PMenu { get; set; }
        public virtual ICollection<Article>? Articles { get; set; }
        public virtual ICollection<Product>? Products { get; set; }
    }
}

File: App/src/app/core/model/menu.ts
import {Article} from "./article";
import {Product} from "./product";

export interface Menu {
  id: number;
  parentMenu: number;
  group: string;
  name: string;
  alias: string;
  index: number;
  showHomePage: boolean;
  type: string;
  active: boolean;
  parentMenuName: string;

  subMenus: Menu[];

  products: Product[],
  articles: Article[]
}
